### PR TITLE
Do not format or add on pre-commit, just fail and let the dev do it

### PR DESCRIPTION
**Pull request by davesteinberg**
The husky changes in PR #111 really threw me for a loop. It's very common for me to make a commit without staging everything, mostly when I'm shuffling bits of work around among different in-progress commits. Or just amending a previous commit to change the message while leaving some new changes unstaged. Having `git add .` in the pre-commit hook is really incompatible with how I work.

I suppose the change was made because if it doesn't make sense to have eslint fix your errors and prettier reformat your code and then not commit those changes.

This PR undoes that. Instead of fixing and reformatting, it just fails if there's a problem. I've renamed the existing `lint` script to `lint-fix` and added a new `lint` that doesn't fix, which is what's now being run in the pre-commit hook.

The idea is that if the hook fails, the commit is blocked, and the developer can run `lint-fix`, verify what it did, stage the changes, and retry the commit. Is this workflow okay with everyone, or are folks really attached to automatic fixing and staging?

### CODE REVIEW

File: lib/prisma.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ declare global {
   // eslint-disable-next-line @typescript-eslint/no-namespace
   namespace NodeJS {
     interface Global {
-      prisma: any
+      prisma?: PrismaClient
```
Comment (davesteinberg) [inline lib/prisma.ts:13]: When I disabled `--quiet` in `lint`, I saw this one warning, so I fixed it.

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+loglevel=silent
```
Comment (davesteinberg) [inline .npmrc:1]: This suppresses the useless npm noise when lint (or any other script) fails.

**Comment by JamiesonRoberts** [comment]
@davesteinberg Yeah I agree with this. I get the original idea behind it, but doing a `git add .` should be an action by the user, not an automated thing.

File: .npmrc (hunk 2 above)
Comment (JamiesonRoberts) [reply .npmrc:1]: Is there any danger in doing this? Like are there any benefits to keeping the NPM noise?

**Review by mwickett — APPROVED** [review]
This is a good change. Definitely agree that adding all doesn't make sense. Sorry gents!

**Review by silent1mezzo — COMMENTED** [review]
Hmm, I agree with not adding automatically but just failing lint and requiring an extra command doesn't make sense to me. Why not meet in the middle, auto fix but then not automatically add?

File: .npmrc (hunk 2 above)
Comment (davesteinberg) [reply .npmrc:1]: In my experience, no. It's a lot of words telling you that something has failed but it's probably not npm's fault, an inverted "npm ERR!" nine times to let you know it's really important, and a pointer to an npm log file that tells you about npm but nothing about the script that failed. And all of this pushes the useful output from the script up and away from your eyes.

I suppose that if there were actually a problem with npm itself, the noise might lead you to look in the log which could help you figure that out, but in the 6 years I've been doing JavaScript development with loglevel=silent, that hasn't ever happened to me.

**Comment by davesteinberg** [comment]
> Hmm, I agree with not adding automatically but just failing lint and requiring an extra command doesn't make sense to me. Why not meet in the middle, auto fix but then not automatically add?

Because that will just commit the unfixed files, which I don't think we want, either.

**Comment by davesteinberg** [comment]
It occurs to me that I didn't mention @LuckeeDev on this. Apologies, I should have. What do you think of this change?

It also occurs to me that if anyone prefers the auto fix and auto add behaviour, they could create an alias to do it:

```
alias commit="npm lint-fix && git add . && git commit"
```

**Comment by davesteinberg** [comment]
Ooo, seems I somehow broke the fonts with this. I'm investigating on PR #166.

**Comment by davesteinberg** [comment]
Okay, it seems I didn't actually break the fonts, it was just some transient Vercel weirdness. I'm going to merge this while things are working again.

**Pull request merged**